<?xml version="1.0"?>
<launch>

    <arg name="namespace" default="bebop" />
    <arg name="ip" default="192.168.42.1" />
    <arg name="config_file" default="$(find bebop_driver)/config/defaults.yaml" />
    <arg name="camera_info_url" default="package://bebop_driver/data/bebop_camera_calib.yaml" />
    <arg name="add_tf_prefix" default="false" />
    <arg name="publish_odom_tf" default="true" />

    <group ns="$(arg namespace)">

        <node pkg="bebop_driver" name="bebop_driver" type="bebop_driver_node" output="screen">
            <param name="camera_info_url" value="$(arg camera_info_url)" />
            <param name="bebop_ip" value="$(arg ip)" />
            <param name="camera_frame_id" value="$(arg namespace)/camera_optical" if="$(arg add_tf_prefix)"/>
            <param name="odom_frame_id" value="$(arg namespace)/odom" if="$(arg add_tf_prefix)"/>
            <param name="base_link_frame_id" value="$(arg namespace)/base_link" if="$(arg add_tf_prefix)" />
            <rosparam command="load" file="$(arg config_file)" />
        </node>

        <include file="$(find bebop_description)/launch/description.launch">
            <arg name="robot_name" value="$(arg namespace)" if="$(arg add_tf_prefix)" />
        </include>

        <include file="$(find bebop_localization)/launch/localization.launch">
            <!-- <arg name="tf_prefix" value="$(arg namespace)" if="$(arg add_tf_prefix)" /> -->
        </include>

    </group>

</launch>
